lib/handlers/subscribe.js

```diff
@@ -60,6 +60,21 @@ function storeSubscriptions (sub, done) {
   var broker = client.broker
   var perst = broker.persistence
 
+  if (sub && packet.subscriptions[0].topic !== sub.topic) {
+    // don't call addSubscriptions per topic,
+    // TODO change aedes subscribe handle, but this is a fast & dirty fix for now
+    return done(null, sub)
+  }
+
+  if (!packet.messageId) {
+    // this is a restoreSubscription, don't addSubscriptions!
+    // TODO change aedes subscribe handle, but this is a fast & dirty fix for now
+    if (perst._addedSubscriptions) {
+      perst._addedSubscriptions(client, packet.subscriptions)
+    }
```

Comment:
I was -100 :))
I should've introduced a `subscriptionsRestored` or `subscriptionsAdded` hook inside aedes' subscribe handler. Then, aedes-cached-persistence would implement it via `_addedSubscriptions`. is it OK?